packages/commonwealth/client/scripts/hooks/api/useTopics.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { useState } from 'react';
+import { Topic } from 'models';
+import { useMutation, useQueryClient, useQuery } from '@tanstack/react-query';
+import $ from 'jquery';
+import app from 'state';
+import queryKeys from './queryKeys';
+
+const useTopics = () => {
+  // auth state to come from zustand, localstorage is just for demo purposes
+  const JWT = localStorage.getItem('jwt');
+
+  const queryClient = useQueryClient();
+  const [topicsForChainId, setTopicsForChainId] = useState<string>();
+
+  const fetchTopicsQuery = useQuery({
+    queryKey: [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+    queryFn: async () => {
+      return queryClient.getQueryData([
```

Comment:
this was a workaround (forgot to comment earlier). I couldn't find the get topics route (maybe the data was coming from `/bulkOffChain`), but that seems related to init flow for the community.

But yes here it will give `undefined` when not cached earlier.